test/replication-luatest/gh_7512_raft_leader_hang_test.lua

```diff
@@ -0,0 +1,67 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+
+local fio = require('fio')
+
+local g = t.group('gh-7512')
+
+
+-- gh-7512: Raft followers should notice leader hang and arrange new elections.
+g.before_all(function(cg)
+    cg.cluster = cluster:new{}
+    local box_cfg = {
+        election_mode = 'candidate',
+        replication_timeout = 0.1,
+        replication_synchro_quorum = 1,
+        replication = {
+            server.build_instance_uri('node1'),
+            server.build_instance_uri('node2'),
+        },
+    }
+    cg.node1 = cg.cluster:build_and_add_server{
+        alias = 'node1',
+        box_cfg = box_cfg,
+    }
+    box_cfg.election_mode='voter'
+    cg.node2  = cg.cluster:build_and_add_server{
+        alias = 'node2',
+        box_cfg = box_cfg,
+    }
+    cg.cluster:start()
+
+    cg.unique_filename = server.build_instance_uri('unique_filename')
+    fio.unlink(cg.unique_filename)
+end)
+
+g.after_all(function(cg)
+    cg.cluster:drop()
+    fio.unlink(cg.unique_filename)
+end)
+
+local function block(name)
```

Comment:
Nit: consider renaming `name` to something like `filename`.

Reply:
Ok, see diff above